syntax = "proto3";

package device;

service DeviceService {
  rpc GetInfo(Empty) returns (DeviceInfo);
  rpc GetState(Empty) returns (DeviceState);
  rpc SetState(DeviceState) returns (CommandResponse);
}

message DeviceInfo {
  string device_name = 1;
  string device_ip = 2;
  string device_port = 3;
  string device_status = 4;
  string device_type = 5; // sensor | atuador
  map<string, string> parametros = 6;
}

message DeviceState {
  string status = 1;       // ativo | desativado
  map<string, string> parameters = 2;
}

message DeviceStateUpdate {
  string device_name = 1;
  string status = 2;
  map<string, string> parameters = 3;
}

message CommandResponse {
  string status = 1;      // ok | error
  string message = 2;
}

message Empty {}
